In [1]:
import pandas as pd

# Load data using relative paths
brands = pd.read_csv('../data/raw/brands.csv')
orderlines = pd.read_csv('../data/raw/orderlines.csv')
orders = pd.read_csv('../data/raw/orders.csv')
products = pd.read_csv('../data/raw/products.csv')

# Create working copies
brands_df = brands.copy()
orderlines_df = orderlines.copy()
orders_df = orders.copy()
products_df = products.copy()

In [2]:
orderlines_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 293983 entries, 0 to 293982
Data columns (total 7 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   id                293983 non-null  int64 
 1   id_order          293983 non-null  int64 
 2   product_id        293983 non-null  int64 
 3   product_quantity  293983 non-null  int64 
 4   sku               293983 non-null  object
 5   unit_price        293983 non-null  object
 6   date              293983 non-null  object
dtypes: int64(4), object(3)
memory usage: 15.7+ MB


### No null values in the dataframe

In [3]:
orderlines_df.describe()

,id,id_order,product_id,product_quantity
count,2.939830e+05,293983.000000,293983.0,293983.000000
mean,1.397918e+06,419999.116544,0.0,1.121126
std,1.530096e+05,66344.486479,0.0,3.396569
min,1.119109e+06,241319.000000,0.0,1.000000
25%,1.262542e+06,362258.500000,0.0,1.000000
50%,1.406940e+06,425956.000000,0.0,1.000000
75%,1.531322e+06,478657.000000,0.0,1.000000
max,1.650203e+06,527401.000000,0.0,999.000000


### All product_ids are 0

In [4]:
orderlines_df['product_quantity'].describe()

count    293983.000000
mean          1.121126
std           3.396569
min           1.000000
25%           1.000000
50%           1.000000
75%           1.000000
max         999.000000
Name: product_quantity, dtype: float64

In [5]:
orderlines_df['product_quantity'].nlargest(20)

53860     999
68712     999
57796     800
57306     555
40813     201
136675    200
204637    192
246048    164
27779     126
285492    125
31113     100
70916     100
31668      99
72394      99
39923      98
40465      97
8578       92
222563     89
7978       80
107045     72
Name: product_quantity, dtype: int64

## Documentation

### Dataset Overview
- **~300,000 rows** with **7 columns**
- No null values in the dataframe
- All `product_id` values are 0 (data quality issue)

### Key Findings
- Over 75% of all orders have `product_quantity == 1`
- Some significant outliers exist with quantities ranging from 555 to 999